src/plugins/data_source/server/routes/data_source_connection_validator.ts

```diff
@@ -36,26 +38,42 @@ export class DataSourceConnectionValidator {
     }
   }
 
-  async fetchDataSourceVersion() {
-    let dataSourceVersion = '';
+  async fetchDataSourceInfo() {
+    const dataSourceInfo: DataSourceInfo = {
+      dataSourceVersion: '',
+      dataSourceEngineType: DataSourceEngineType.NA,
+    };
+
     try {
       // OpenSearch Serverless does not have version concept
       if (
         this.dataSourceAttr.auth?.credentials?.service === SigV4ServiceName.OpenSearchServerless
       ) {
-        return dataSourceVersion;
+        dataSourceInfo.dataSourceEngineType = DataSourceEngineType.OpenSearchServerless;
+        return dataSourceInfo;
       }
+
       await this.callDataCluster
         .info()
         .then((response) => response.body)
         .then((body) => {
-          dataSourceVersion = body.version.number;
+          dataSourceInfo.dataSourceVersion = body.version.number;
+
+          if (
+            body.version.distribution !== null &&
+            body.version.distribution !== undefined &&
+            body.version.distribution === 'opensearch'
```

Comment:
```suggestion
          if (
            body.version.distribution === 'opensearch'
```
This can be simplified